job-server/src/main/scala/spark/jobserver/WebApi.scala

```diff
@@ -514,10 +515,47 @@ class WebApi(system: ActorSystem,
    *    GET /healthz              - return OK or error message
    */
   def healthzRoutes: Route = pathPrefix("healthz") {
+    import ContextSupervisor._
+    import JobManagerActor._
+    import JobInfoActor._
+    import JobDAOActor._
     //no authentication required
     get { ctx =>
       logger.info("Receiving healthz check request")
-      ctx.complete("OK")
+      var actorsAlive: Int = 0
+      val supervisorFuture = (supervisor ? GetContext("dummycontext")).mapTo[NoSuchContext.type]
+      val jobInfofuture = (jobInfoActor ? GetJobStatus("dummyjobid")).mapTo[NoSuchJobId.type]
+      val jobDaofuture = (daoActor ? GetJobInfos(1)).mapTo[JobInfos]
+      val statusActor = system.actorOf(JobStatusActor.props(daoActor))
+      val jobStatusfuture = (statusActor ? Unsubscribe("dummyjobid", null)).mapTo[NoSuchJobId.type]
```

Comment:
I am not sure I see the point of creating statusActor and then sending a message to it here. I would say these two line should be removed.

Reply:
Can you please elaborate on why this is not so useful?

Reply:
ping @valan4ik

Reply:
If I understand it correctly, you create a new status actor here and try to send him a message.
But this actor is just artificially created here and is not part of the live system.
I am not sure I get the idea and what value does it bring, if the request fails it doesn't say anything about current cluster status.
In case of other actors, you check the ones, which form the akka cluster (like supervisor, which is a core of jobserver and of course if he is not here, system is unhealthy).

Reply:
Taken care of this